engine/consensus/matching/engine.go

```diff
@@ -447,6 +447,13 @@ func (e *Engine) storeReceipt(receipt *flow.ExecutionReceipt, head *flow.Header)
 	if err != nil && !errors.Is(err, storage.ErrAlreadyExists) {
 		return false, fmt.Errorf("could not persist receipt: %w", err)
 	}
+
+	// index receipt by executor
+	err = e.indexReceipt(receipt)
```

Comment:
I don't think we need to index the receipt that hasn't been added to a block's payload yet. 

We might want to index the receipt when `receiptsDB.Store` is called, so that the receipt and its index will be stored in one transaction.

The reason the indexing is not added in the `Store` method is that not every node cares about this index. 

Therefore, on master, [we let the Indexer module dedicated for this job](https://github.com/onflow/flow-go/blob/master/engine/consensus/matching/indexer.go#L45). If we also index receipts here, I felt the Indexer module will be sad that it's not clear who is responsible for indexing receipts. 

It's OK even if we don't index receipt here, we weren't indexing receipts that are included in non-finalized blocks. 

In the long term, I think it's easier to index receipts when we store them, we just need to think clearly about the best way to index them.